compiler-core/src/language_server/completer.rs

```diff
@@ -25,6 +25,29 @@ use super::{
     compiler::LspProjectCompiler, files::FileSystemProxy, DownloadDependencies, MakeLocker,
 };
 
+#[derive(Copy, Clone)]
+enum SortPriority {
+    // A label for a function or type definition
+    Label,
+    // A field of a record
+    FieldAccessor,
+    // Values or types defined in the current module
+    LocallyDefined,
+    // Values or types defined in an already imported module
+    ImportedModule,
+    // Types defined in the prelude
+    Prelude,
+    // Types defined in a module that has not been imported
+    ImportableModule,
+}
+
+// Gives the sort text for a completion item based on the priority and label.
+// This ensures that higher priority items are placed before lower priority items.
+fn sort_text(priority: SortPriority, label: &str) -> String {
+    let priority = priority as u8;
```

Comment:
This seems a bit cryptic and error prone. Perhaps a big bold comment is needed on the definition to say that the ordering matter, which is a unique property for this type.

Reply:
Yeah totally fair, I added a nice doc comment to the type definition. lmk if that's enough or if i should add something here as well